cowprotocol/accounting/rewards/mainnet/slippage_query_3427730.sql

```diff
@@ -0,0 +1,162 @@
+            date_trunc('hour', block_time) as hour,
+            usd_value / units_sold as price
+        from cow_protocol_ethereum.trades
+        where
+            block_number >= (select start_block from block_range) and block_number <= (select end_block from block_range)
+            and units_sold > 0
+    ) as combined
+    group by hour, contract_address, decimals
+    order by hour
+),
+
+-- -- Price Construction: https://dune.com/queries/1579091?
+prices as (
+    select
+        tt.hour,
+        tt.token as contract_address,
+        coalesce(
+            precise.decimals,
+            intrinsic.decimals
+        ) as decimals,
+        coalesce(
+            precise.price,
+            intrinsic.price
+        ) as price
+    from token_times as tt
+    left join precise_prices as precise
+        on
+            tt.hour = precise.hour
+            and precise.contract_address = token
+    left join intrinsic_prices as intrinsic
+        on
+            tt.hour = intrinsic.hour
+            and intrinsic.contract_address = token
+),
+
+-- -- ETH Prices: https://dune.com/queries/1578626?d=1
+eth_prices as (
+    select
+        date_trunc('hour', minute) as hour,
+        avg(price) as eth_price
+    from prices.usd
+    where
+        blockchain = 'ethereum'
+        and contract_address = 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+        and minute between cast('{{start_time}}' as timestamp) and cast('{{end_time}}' as timestamp)
+    group by date_trunc('hour', minute)
+),
+
+results_per_tx as (
+    select
+        ftbs.hour,
+        tx_hash,
+        solver_address,
+        sum(cast(token_imbalance_wei as double) * price / pow(10, p.decimals)) as usd_value,
+        sum(cast(token_imbalance_wei as double) * price / pow(10, p.decimals) / eth_price) * pow(10, 18) as eth_slippage_wei,
+        count(*) as num_entries
+    from
+        final_token_balance_sheet as ftbs
+    left join prices as p
+        on
+            token = p.contract_address
+            and ftbs.hour = p.hour
+    left join eth_prices as ep
+        on ftbs.hour = ep.hour
+    group by
+        ftbs.hour,
+        solver_address,
+        tx_hash
+    having
+        bool_and(price is not null)
+),
+
+results as (
+    select
+        solver_address,
+        concat(environment, '-', name) as solver_name,
+        sum(usd_value) as usd_value,
+        sum(eth_slippage_wei) as eth_slippage_wei,
+        concat(
+            '<a href="https://dune.com/queries/3427730?SolverAddress=',
+            cast(solver_address as varchar),
+            '&CTE_NAME=results_per_tx',
+            '&StartTime={{start_time}}',
+            '&EndTime={{end_time}}',
+            '" target="_blank">link</a>'
+        ) as batchwise_breakdown
+    from
+        results_per_tx
+    inner join cow_protocol_ethereum.solvers
+        on address = solver_address
+    group by
+        solver_address,
+        concat(environment, '-', name)
+)
+
+select * from {{cte_name}}
```

Comment:
this parameter seems unnecessary

Reply:
this is so that we can also see the `results_per_tx` table, if needed.

Reply:
Did some changes to address some of these.